src/bigint.cpp

```diff
@@ -1447,9 +1410,12 @@ void bigint_negate_wrap(BigInt *dest, const BigInt *op, size_t bit_count) {
 }
 
 void bigint_not(BigInt *dest, const BigInt *op, size_t bit_count, bool is_signed) {
-    if (bit_count == 0) {
-        bigint_init_unsigned(dest, 0);
-        return;
+    if (!bit_count) {
+        bit_count = (op->digit_count * 64) - bigint_clz(op, op->digit_count * 64) + (is_signed ? 1 : 0);
```

Comment:
why this change?

this function is supposed to evaluate binary not based on the bit count passed in as a parameter. So if the passed in value is 0 then binary not of 0 is 0.

Reply:
@andrewrk I will document this function.

> So if the passed in value is 0 then binary not of 0 is 0.

I think that is an incorrect assumption -- `bit_count` only specifies how many bits should be included in the `not` operation; furthermore, `BigInt` is architected only to store a value. zig seems to have a u0 type, but I don't see how this is valuable or used.

This implementation correctly returns `not` of `op` into `dest` with optional bound of `bit_count`.